File: CSharpPractice/LeetcodePractice/Leetcode124.cs
public class Solution124 {
    int ans=-10000;
    public int helper(TreeNode node)
    {
        if(node==null){return 0;}
        int leftSum = Math.Max(0,helper(node.left));
        int rightSum = Math.Max(0,helper(node.right));
        int tempAns = node.val + leftSum + rightSum;
        ans=Math.Max(ans,tempAns);
        return node.val + Math.Max(leftSum,rightSum);
    }
    public int MaxPathSum(TreeNode root) 
    {
        helper(root);
        return ans;
    }
}

File: JavaScriptPractice/Leetcode124.js
let ans=0;
var helper = function(node)
{
    if(node==null){return 0 ;}
    let leftSum = Math.max(0,helper(node.left));
    let rightSum = Math.max(0,helper(node.right));
    let tempAns = node.val + leftSum + rightSum;
    ans =Math.max(ans,tempAns);
    return node.val + Math.max(leftSum,rightSum);
};
var maxPathSum = function(root) 
{
    ans=-10000;
    helper(root);
    return ans;
};